### DISCUSSION TITLE
Building for Windows

### DISCUSSION THREAD

**Discussion by seastones** [DISTRHO/Cardinal#469] [Q&A]
So the other day I took to adding a few modules for personal use and testing, and had quite some success building in Manjaro. I'm primarily a Windows user however, and many of the tools I use in making music are locked to Windows so the intention was to build for Windows. This is where things get weird...

So the native build in Manjaro compiled fine and ran.
Trying to build with mingw was a no go, with what I recall were errors stemming from mingw itself. (Vaguely I remember many notes about pthreads and finally an issue with mutex.)
Following that failure, I took a look at the Actions and saw that the build process for Windows was very different and used Ubuntu. Figuring it might be an issue with the version of mingw (as they were different), I set up Ubuntu and gave the process as outlined by the Actions a shot. Things got weirder...
The Windows build was failing when creating the CLAP plugin, with ODE errors seeming to stem from SurgeXT components(if I'm recalling properly, I'm not at the computer at the moment to check).
Just to sanity check that it would compile, I then tried to build natively in the same manner I had success with in Manjaro. This also failed, but with a complaint about plugins-mini.a and a version mismatch. (9.2 vs 11.2 or some such, I believe coming from ld).

I suppose the question is a general "where did I go wrong?" More specifically, are there any peculiarities to building for Windows?

Thank you in advance!

**Comment by falkTX** [comment]
No peculiarities, though with the addition of JUCE things got a bit more complicated.
You should start trying to build using `make NOPLUGINS=true` to ensure it can even build the most basic things.

Also, dont bother try building from latest git, I do my tests and incomplete things directly on the main branch.
Use a release tag if you dont want surprises, the new work for a mini version is not yet complete, build failures are expected at the moment with latest git.

**Reply by seastones** [reply]
I'll definitely give that a shot. I'm pretty sure I checked the commit I pulled for a successful build for win64, but I'm not sure how much that helps.

I can also provide logs later if I run into more trouble.
